### DISCUSSION TITLE
PRACH configuration

### DISCUSSION THREAD

**Discussion by ThorisoMakhosa** [srsran/srsRAN_Project#167] [General Help] [answered]
Hi experts your assistance will be much appreciated.

May you please assist with the understanding of the log file attached below particularly for the initial RACH signaling, I am only looking at the physical channels.
From my understanding of the RACH process, the UE starts by transmitting a random Zadoff-Chu sequence which is correlated with all the possible sequences and once the transmitted zadoff-chu sequence is determined the gNB then responds, allocating the RA-RNTI and Uplink grant through the scheduler and the PDCCH including information such as the slot fomart,etc. Then after this the UE can transmit uplink data on the PUSCH. 
However on the log files, there is information transmitted on the PDSCH before the UE can transmit information on the PUSCH and i don't understand how such is the case, I thought the UE is suppose to send the user plane data first.

**Comment by ThorisoMakhosa** [comment]
![log](https://github.com/srsran/srsRAN_Project/assets/107929145/3a1046de-73f4-45f1-8aae-eaa929691ea6)
 
I have included a screenshot which i believe includes the initial RACH logs since the log file itself cant be attached because its very huge

**Reply by frankist** [reply, marked as answer]
Hey! I explain below the different steps involved in the random access:

- slot=145.4: PRACH received. From the PRACH we can derive the RA-RNTI.
- slot=145.9: RAR sent. It requires both PDCCH and PDSCH. RA-RNTI derived from the PRACH is used as RNTI. The RAR PDSCH contains the TC-RNTI that the UE will use for Msg3
- slot=146.5: Msg3 Received. In this case, it contains the RRC Setup request. Note: Msg3 doesn't require a PDCCH as all the information needed for the UE to know when (slot) and where (RBs) to transmit the Msg3 is already contained in the RAR PDSCH payload.
- slot=147.1: Msg4 sent. PDCCH and PDSCH are required. The PDSCH contains the CON_RES MAC CE and the SRB0 PDU with the RRC Setup in it. This completes the Random Access procedure.

It seems to me that you forgot that the RAR uses PDSCH.

**Reply by ThorisoMakhosa** [reply]
Hi. This puts everything into perspective , thank you very much.

**Comment by rainey19** [comment]
I've never heard the thing about UE transmitting UP data first before. If you enable MAC PCAP and then open it in wireshark, you can see what's actually being sent over each channel.

**Reply by rainey19** [reply]
That PDSCH message could be SIB1.

**Reply by frankist** [reply]
There are only two PDSCHs in the log shared above: the RAR and the Msg4.
